exercises/bracket-push/example.ml

```diff
@@ -0,0 +1,26 @@
+open Core.Std
+
+type 'a stack = 'a list
+
+let push (s: 'a stack) (a: 'a): 'a stack = a :: s
+let pop (s: 'a stack): ('a * 'a stack) option = match s with
+  | [] -> None
+  | x :: xs -> Some (x, xs)
+
+let foldThroughOptionMonad (xs: 'a list) (z: 'b) ~(f:'b -> 'a -> 'b option): 'b option =
```

Comment:
Although `Option` is a monad, I am not sure if I like to see it in a name. First of, this is an `example.ml`, so one can argue if we should scrutinize on this level. I do think it is worthwhile because people could end up visiting the repository to glimpse a hint of a solution.

Below I try to formulate the reason I don't like to add `Monad` to a name, but I am not sure if I can confidently relay my vague feelings.

1. You don't have to know that Option is a monad in order to use it. Mentioning it could trigger a congnitive jarring sensation, distraction from the goal of learning a few tricks of the oCaml trade.
2. Monads have quickly become an elitist differentiate, easily alienating new comers. I.e. the myth that monads are hard is perpetrated without explaining them.

The above ramblings does not imply anything about the implementation. I actually really like the implementation. It is a very clear and clever implementation of a stack. Especially the way you implemented the fold is very well done